compiler/rustc_middle/src/ty/consts/int.rs

```diff
@@ -109,3 +113,228 @@ impl std::fmt::Debug for ConstInt {
         }
     }
 }
+
+// FIXME: reuse in `super::int::ConstInt` and `Scalar::Bits`
+/// The raw bytes of a simple value.
+#[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
+#[derive(HashStable)]
```

Comment:
Because `data` is stored in the native endian format, this would break the incremental cache when moving from an little endian to big endian compiler or the other way around. Even if the target is the same. I can also imagine that by simultaneously switching compiler byte order and swapping the bytes in the source code, rustc may think that nothing changed.

Reply:
uh... is that actually a thing? Changing the compiler binary and keeping the cache?

Reply:
For as long as it is compiled from the same source I would expect it to work. Even if the target is changed. Integers are encoded in uleb128/sleb128, so they are encoded in an endian and size independent way. https://github.com/rust-lang/rust/blob/f243a2ad904705a1e340a08639dca105605b4175/compiler/rustc_serialize/src/opaque.rs#L247

Reply:
Makes sense. I guess I could also just make the struct packed and simplify all that code.

Reply:
I fixed this